dev/TreeView/TreeViewList.cpp

```diff
@@ -99,6 +99,14 @@ void TreeViewList::OnContainerContentChanging(const winrt::IInspectable& /*sende
     if (!args.InRecycleQueue())
     {
         auto targetItem = args.ItemContainer().as<winrt::TreeViewItem>();
+
+        // "Re-evaluate" the item source binding to trigger a TreeViewItem.OnPropertyChanged(ItemsSourceProperty) call.
+        if (auto itemsSource = targetItem.ItemsSource())
+        {
+            targetItem.ItemsSource(nullptr);
```

Comment:
What happens if you don't set it to nullptr?

Reply:
The current fix is re-doing the steps after a TreeView is created . I'm setting the source to null so that setting the source again will trigger a TreeViewItem::OnPropertyChanged call. This call is then setting up the child nodes (as TreeViewNode.m_itemsDataSource also seems to be lost upon recycling and needs to be set again). If I don't set it to nullptr here, I would have to set it to something else to trigger the OnPropertyChanged call.

There might be a better way to fix this issue (and I'm also seeing some tests failing locally for me). For now I only found this attempt though.

Reply:
I see, you can probably wrap the [code](https://github.com/microsoft/microsoft-ui-xaml/blob/master/dev/TreeView/TreeViewItem.cpp#L327-L334) in OnPropertyChanged into a separate function and call that function instead, that way you don't have to set ItemsSource to null first just to trigger OnPropertyChanged.

Reply:
Done.